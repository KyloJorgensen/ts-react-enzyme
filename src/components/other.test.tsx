'use strict';

import * as React from 'react';
import { shallow } from 'enzyme';
import Other from './other';

describe('Render <Other/> component', () => {
	it('matches the snapshot', () => {
		const wrapper = shallow(<Other />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should have only one div element', () => {
		const wrapper = shallow(<Other />);
		const div = wrapper.find('div');
		expect(div).toHaveProperty('length', 1);
	});

	it('should have a div with class of other', () => {
		const wrapper = shallow(<Other />);
		const otherClass = wrapper.find('div').hasClass('other')
		expect(otherClass).toBeTruthy();
	});

	it('should have only one p element', () => {
		const wrapper = shallow(<Other />);
		const p = wrapper.find('p');
		expect(p).toHaveProperty('length', 1);
	});

	it('should have a p element with text', () => {
		const wrapper = shallow(<Other/>);
		const p = wrapper.find('p');
		const text = p.text();
		expect(text).toEqual('Other Page');
	});
});
