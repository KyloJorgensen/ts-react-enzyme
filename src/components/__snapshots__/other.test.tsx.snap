// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <Other/> component matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Other />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Other Page
      </p>,
      "className": "other",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Other Page",
      },
      "ref": null,
      "rendered": "Other Page",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Other Page
        </p>,
        "className": "other",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Other Page",
        },
        "ref": null,
        "rendered": "Other Page",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
